src/worker.rs

```diff
@@ -138,9 +138,18 @@ pub(crate) async fn worker_main(goose_attack: GooseAttack) -> GooseAttack {
         if worker_id == 0 {
             worker_id = initializer.worker_id;
         }
+        // If no host was specified in the initializer, use the host from the worker configuration.
         let user = GooseUser::new(
             initializer.scenarios_index,
-            Url::parse(&initializer.base_url).unwrap(),
+            Some(
+                Url::parse(
+                    initializer
+                        .base_url
+                        .unwrap_or(goose_attack.configuration.host.clone())
+                        .as_str(),
+                )
```

Comment:
this looks a little odd. Using `unwrap_or` is good however the need to clone and then reference the cloned value seems off. Why not just reference the value to begin with instead of cloning?

Does this change work? Maybe there is a lifetime issue I am not spotting in my read of the code.
```suggestion
                Url::parse(
                    initializer
                        .base_url
                        .unwrap_or(goose_attack.configuration.host.as_str()),
                )
```

Reply:
Actually, this didn't work... `unwrap_or` expects to take ownership of the default value.

Since `base_url` is an `Option<String>`, `unwrap_or` thus expects to consume a `String` instead of accepting a reference to it.
```
    pub const fn unwrap_or(self, default: T) -> T
    where
        T: ~const Destruct,
    {
        match self {
            Some(x) => x,
            None => default,
        }
    }
 ```

Reply:
If `clone` is indeed the answer, can/should I clone it outside the loop? Right now the `clone` will happen in each iteration of the loop.

Reply:
Ok in that case lets make the Option a reference first:


```
              Url::parse(
                    initializer
                        .base_url
                        .as_ref() // Option<&String>
                        .unwrap_or(goose_attack.configuration.host.as_str()),
                )
```

or if that doesn't work we can use `.map(|url| url.as_str())` and then `unwrap_or`

Reply:
ooo I like that! Just pushed.